python/ray/autoscaler/_private/_azure/node_provider.py

```diff
@@ -107,7 +107,11 @@ def match_tags(tags):
             k: v for k, v in self.terminating_nodes.items() if not v.done()
         }
 
-        return {k: v for k, v in self.cached_nodes.items() if match_tags(v["tags"])}
+        return {
+            k: v
+            for k, v in self.cached_nodes.items()
+            if v["tags"] is not None and match_tags(v["tags"])
```

Comment:
Do we need to check if "tags" key exists?

Reply:
Yeah, it is to fix #48976 issue.